### PR TITLE
Return the service cluster for HyperShift clusters with `ocm describe cluster`

### PR DESCRIPTION
**Pull request by mjlshen**
When troubleshooting HyperShift clusters, sometimes it is important to inspect the relevant service cluster. This PR does this by essentially implementing:

```
❯ ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="id='$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="name='hs-mc-op846kea0'" | jq -r '.items[0].parent.id')'" | jq -r '.items[0].name'

hs-sc-op846kde0
```

Sample output:
```
❯ ocm describe cluster 21rrnvvf6rd4vdndl4lf3gie7mat0gna                                

ID:			21rrnvvf6rd4vdndl4lf3gie7mat0gna
External ID:		d61a016b-dd48-4f96-ac27-dc9a1b1f7b63
...
Management Cluster:     hs-mc-jsui4ri9g
Service Cluster:        hs-sc-op846kde0

```

### CODE REVIEW

File: pkg/cluster/describe.go [hunk 1]
```diff
@@ -158,15 +159,30 @@ func PrintClusterDescription(connection *sdk.Connection, cluster *cmv1.Cluster)
...
+
+		mgmtClusterId = fmMgmtResp.Items().Get(0).ClusterManagementReference().ClusterId()
+
+		// href is in the format /api/osd_fleet_mgmt/v1/service_clusters/${id}
+		// and we are using it because ocm-sdk-go currently does not export the id directly
+		if kind := fmMgmtResp.Items().Get(0).Parent().Kind(); kind == "ServiceCluster" {
+			href := fmMgmtResp.Items().Get(0).Parent().Href()
+			_, id := path.Split(href)
+			fmSvcResp, _ := connection.OSDFleetMgmt().V1().ServiceClusters().ServiceCluster(id).Get().Send()
+			svcClusterId = fmSvcResp.Body().ClusterManagementReference().ClusterId()
```
Comment (mjlshen) [inline pkg/cluster/describe.go:184]: The main reason I switched the management cluster to show its internal ID instead of the name as it was previously was due to the cluster name of the service cluster not being available here - I'd have to make an additional call to ClustersMgmt in order to get the name of a service cluster.

Happy to do so if that's preferred, but I felt that it was ok to show the ID instead since those are guaranteed to be unique.

cc @fahlmant

Comment (fahlmant) [reply pkg/cluster/describe.go:184]: Either way works, this is for the SRE mostly, so its up to us to decide. We're already making multiple API calls, so I don't think one more would hurt, and reasoning about a name like `hs-mc-xxxxx` is a bit easier than an ID, but we use both constantly and internal ID is unique.

tl;dr either way works, there's an argument for both ways being a better UX so I have no idea what we should do

Comment (mjlshen) [reply pkg/cluster/describe.go:184]: Got it - yeah agreed that `hs-mc-xxxx`/`hs-sc-xxxx` is easier to grok if this is mostly for SRE. I'll make that change

Comment (gdbranco) [reply pkg/cluster/describe.go:184]: Is there anyway to enforce this only shows up for specific roles/permissions taken from token? Don't think usual clients will get much from this info

Comment (fahlmant) [reply pkg/cluster/describe.go:184]: During printing, we already do this check 
```
if svcClusterId != "" {
```

Comment (mjlshen) [reply pkg/cluster/describe.go:184]: also https://github.com/openshift-online/ocm-cli/pull/441/files#diff-2f84bc16b5d89dac13d11d6f8ae66c242bc75cccb194fdd0ace7723bb75b9312R165 we ignore errors and don't print anything by default - so it will only print if the specific roles/permissions are in place

File: pkg/cluster/describe_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	tests := []struct {
+		name         string
+		cluster      *cmv1.Cluster
+		expectedMgmt string
+		expectedSvc  string
+	}{
+		{
+			name:    "Not HyperShift",
+			cluster: newTestCluster(t, cmv1.NewCluster().Hypershift(cmv1.NewHypershift().Enabled(false))),
+		},
```
Comment (mjlshen) [inline pkg/cluster/describe_test.go:28]: The current test-case covers the really simple case where this is not a HyperShift cluster and I'm not sure how to mock `conn *sdk.Connection` so that I can unit test HyperShift clusters. If there are any tips or examples I am happy to do so

File: pkg/cluster/describe.go [hunk 3]
```diff
@@ -272,3 +275,48 @@ func PrintClusterDescription(connection *sdk.Connection, cluster *cmv1.Cluster)
...
+		href := fmMgmtResp.Items().Get(0).Parent().Href()
+		_, id := path.Split(href)
+		fmSvcResp, _ := conn.OSDFleetMgmt().V1().ServiceClusters().ServiceCluster(id).Get().Send()
+		svcClusterId := fmSvcResp.Body().ClusterManagementReference().ClusterId()
+
+		svcClusterResp, err := conn.ClustersMgmt().V1().Clusters().Cluster(svcClusterId).Get().Send()
+		if err != nil {
+			return mgmtClusterName, ""
+		}
+		return mgmtClusterName, svcClusterResp.Body().Name()
```
Comment (vkareh) [inline pkg/cluster/describe.go:317]: Why are you looking up the cluster ID if you're returning the name? That should be available in the MC payload already:
```
if kind := fmMgmtResp.Items().Get(0).Parent().Kind(); kind == "ServiceCluster" {
  svcClusterName := fmMgmtResp.Items().Get(0).Parent().Name()
}
```

Comment (vkareh) [reply pkg/cluster/describe.go:317]: okay, just saw the the SDK doesn't expose the SC name, but it's part of the payload. Can you add this to the corresponding model file? https://github.com/openshift-online/ocm-api-model/blob/main/model/osd_fleet_mgmt/v1/management_cluster_parent_type.model

Comment (mjlshen) [reply pkg/cluster/describe.go:317]: Name isn't exported by ocm-sdk-go :( It's returned when I hit the API via `ocm get` though

File: pkg/cluster/describe.go [hunk 4]
```diff
@@ -272,3 +275,48 @@ func PrintClusterDescription(connection *sdk.Connection, cluster *cmv1.Cluster)
...
+		Get().
+		Send()
+	if err != nil {
+		return "", ""
+	}
+
+	mgmtClusterName := hypershiftResp.Body().ManagementCluster()
+	fmMgmtResp, err := conn.OSDFleetMgmt().V1().ManagementClusters().
+		List().
+		Parameter("name", mgmtClusterName).
```
[the remark this answers is not in the document]
Comment (mjlshen) [reply pkg/cluster/describe.go:299]: This does work, maybe via undefined behavior though :D - happy to change to this if that's how it's supposed to work

[the remark this answers is not in the document]
Comment (tiwillia) [reply pkg/cluster/describe.go:299]: I'm also very surprised this works... what the heck

File: pkg/cluster/describe.go (hunk 3 above)
Comment (mjlshen) [reply pkg/cluster/describe.go]: Added to the model file in https://github.com/openshift-online/ocm-api-model/pull/690 and verified it worked - but now this PR should not be merged until that one is + a corresponding bump in ocm-sdk-go + a dependency bump in ocm-cli. So I am marking this PR as WIP.

Comment (vkareh) [reply pkg/cluster/describe.go]: There's now a new SDK version with support for service/management cluster names. Can you update this?

Comment (mjlshen) [reply pkg/cluster/describe.go]: Done! Thanks for pushing the SDK update forward!

**Comment by vkareh** [comment]
@mjlshen do you still consider this `WIP`?

**Comment by mjlshen** [comment]
Nope, thanks for the reminder - I've removed `WIP`

File: pkg/cluster/describe.go [hunk 5]
```diff
@@ -272,3 +274,37 @@ func PrintClusterDescription(connection *sdk.Connection, cluster *cmv1.Cluster)
...
+	hypershiftResp, err := conn.ClustersMgmt().V1().Clusters().
+		Cluster(cluster.ID()).
+		Hypershift().
+		Get().
+		Send()
+	if err != nil {
+		return "", ""
+	}
+
+	mgmtClusterName := hypershiftResp.Body().ManagementCluster()
```
Comment (igoihman) [inline pkg/cluster/describe.go:295]: what will be the output of this command for users who can't access the management cluster details?
